modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/forms/input/InputLocalized.tsx

```diff
@@ -80,20 +71,37 @@ export default function InputLocalized({
 	placeholder,
 	required,
 	resultFormatter = () => null,
-	selectedLocale,
+	selectedLocale: initialSelectedLocale,
 	tooltip,
 	translations: initialTranslations,
 	...otherProps
 }: InputLocalizedProps) {
-	const [locale, setLocale] = useState<InputLocale>(availableLocales[0]);
+	const availableLocales = useMemo(() => {
+		return Object.keys(Liferay.Language.available)
+			.sort((languageId: string) =>
+				languageId === Liferay.ThemeDisplay.getDefaultLanguageId()
+					? -1
+					: 1
+			)
+			.map((languageId: string) => ({
+				label: languageId as Liferay.Language.Locale,
+				symbol: languageId.replace(/_/g, '-').toLowerCase(),
+			}));
+	}, []);
```

Comment:
Used `useMemo` here to only calculate this once since the values shouldn't change on re-render.

Reply:
I would typically use a `useRef` for this purpose. You set it once, and the value is stable on renders. But I don't see why your solution wouldn't work as well. It would make a difference if `useMemo` had a dependency, of if we ever wanted to update `ref` value. But we don't, so these become functionally the same.

Even considering just what makes more readable code, I'm not sure which approach wins. `useRef` seem nicer to me, with touch less boilerplate, but that may be just me :man_shrugging:

Reply:
My thinking is that `useMemo` made more sense here since it makes it clear the value is never updated. With `useRef` that value can change if you set a new value through `.current` so I feel that'd be a more likely use case for `useRef` over `useMemo`.